### PR TITLE
To draw KVB and TVM signals

### PR DESCRIPTION
**Pull request by Sovxx**
To (hopefully, I am not sure what I am doing !) solve #70

### CODE REVIEW

File: signals.mml [hunk 1]
```diff
@@ -110,7 +110,9 @@ Layer:
                 tags->'railway:atb-ng' AS atb_ng,
                 tags->'railway:atb-vv' AS atb_vv,
                 railway_etcs_null_no(tags->'railway:etcs') AS etcs,
-                railway_etcs_null_no(tags->'construction:railway:etcs') AS construction_etcs
+                railway_etcs_null_no(tags->'construction:railway:etcs') AS construction_etcs,
+                tags->'railway:kvb' AS kvb,
+                tags->'railway:tvm' AS tvm
```
Comment (DerDakon) [inline signals.mml:115]: I would move the new tags before the etcs ones, this would reduce the amount of changed lines (but has no technical difference).

Comment (Sovxx) [reply signals.mml]: Thanks, done !

**Review by DerDakon — CHANGES_REQUESTED** [review]
This seems correct to select the data from the database, but you don't do anything with it. You would need to add drawing instructions for the matched lines also to [train_protection.mss](https://github.com/OpenRailwayMap/OpenRailwayMap-CartoCSS/blob/master/train_protection.mss).

**Comment by Sovxx** [comment]
OK, I have just made changes to train_protection.mss
Could you please check ?

**Review by DerDakon — APPROVED** [review]

**Comment by JJJWegdam** [comment]
@Nakaner: a friendly reminder that this PR is waiting for approval. If we start rendering TVM/KVB it would really help me to map it

File: signals.mml [hunk 2]
```diff
@@ -98,8 +98,8 @@ Layer:
             NULL AS construction_usage, NULL AS construction_service,
             NULL AS preserved_railway, NULL AS preserved_service,
             NULL AS preserved_usage,
-            pzb, lzb, atb, atb_eg, atb_ng, atb_vv, etcs, construction_etcs,
-            railway_train_protection_rank(pzb, lzb, atb, atb_eg, atb_ng, atb_vv, etcs, construction_etcs) AS rank
+            pzb, lzb, atb, atb_eg, atb_ng, atb_vv, kvb, tvm, etcs, construction_etcs,
+            railway_train_protection_rank(pzb, lzb, atb, atb_eg, atb_ng, atb_vv, kvb, tvm, etcs, construction_etcs) AS rank
```
Comment (Nakaner) [inline signals.mml:102]: If you add new arguments to the `railway_train_protection_rank` function, you have to add support for those in `sql/functions.sql` as well.

**Review by Nakaner — CHANGES_REQUESTED** [review]

File: signals.mml (hunk 2 above)
Comment (Sovxx) [reply signals.mml:102]: OK, I have changed `sql/functions.sql`
Can you have a look?
Thanks :)

**Comment by Sovxx** [comment]
@Nakaner Hi, please let me know if any more change is needed. Thanks :)

**Comment by Sovxx** [comment]
@Nakaner Hit, is there a way I (or anyone) could help ? Thanks :)

**Comment by DerDakon** [comment]
This needs a rebase. Please also have a look at #81 which tries to do the same thing.

**Comment by Sovxx** [comment]
Too complicated (why rebase ?) and too long for me.
Please help #81 instead.

**Comment by davide84** [comment]
Hi! I'm author of #81 which needs fixes but not a rebase. I will continue on it but double-check with this one to make sure no change is left out. Thanks!

**Pull request closed without merging**